pydoctor/util.py

```diff
@@ -0,0 +1,64 @@
+"""
+Misc utils.
+"""
+import os
+import subprocess
+
+
+def get_source_reference() -> str:
+    """
+    Return a reference to the particular source version used in this build.
+    This can be used in the URL passed with C{--html-viewsource-base}, to link to the
+    version of the source code that corresponds with the generated documentation.
+
+    It tries the extract the tag name from Read The Docs environment and
+    falls back to source code revision.
+
+    Only git VSC is supported for now.
+
+    @raises RuntimeError: If source reference couldn't be retrieved.
+    """
+    if os.environ.get("READTHEDOCS", "") == "True":
+        rtd_version = os.environ.get("READTHEDOCS_VERSION", "")
+        if rtd_version not in ['latest', 'stable']:
```

Comment:
I generally prefer tuples over lists if mutability is not required. Maybe there could be runtime optimizations done for tuples that aren't done for lists, but I don't know if that is actually the case for CPython or pypy.